api/src/test/java/org/openmrs/api/OrderSetServiceTest.java

```diff
@@ -0,0 +1,156 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api;
+
+import org.dbunit.dataset.IDataSet;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.openmrs.OrderSet;
+import org.openmrs.OrderSetMember;
+import org.openmrs.User;
+import org.openmrs.api.context.Context;
+import org.openmrs.test.BaseContextSensitiveTest;
+import org.openmrs.test.Verifies;
+
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class OrderSetServiceTest extends BaseContextSensitiveTest {
+	
+	protected OrderService orderService;
+	
+	protected OrderSetService orderSetService;
+	
+	protected ConceptService conceptService;
+	
+	protected static final String ORDER_SET = "org/openmrs/api/include/OrderSetServiceTest-general.xml";
+	
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
+	
+	/**
+	 * Run this before each unit test in this class. The "@Before" method in
+	 * {@link BaseContextSensitiveTest} is run right before this method.
+	 *
+	 * @throws Exception
+	 */
+	@Before
+	public void runBeforeAllTests() throws Exception {
+		if (null == orderSetService) {
+			orderSetService = Context.getOrderSetService();
+		}
+		if (null == orderService) {
+			orderService = Context.getOrderService();
+		}
+		if (null == conceptService) {
+			conceptService = Context.getConceptService();
+		}
+	}
+	
+	@After
+	public void revertToDefaultLocale() throws Exception {
+		Context.setLocale(Locale.US);
+	}
+	
+	/**
+	 * Updates the search index to clean up after each test.
+	 *
+	 * @see org.openmrs.test.BaseContextSensitiveTest#updateSearchIndex()
+	 */
+	@Before
+	@Override
+	public void updateSearchIndex() {
+		super.updateSearchIndex();
+	}
+	
+	/**
+	 * Updates the search index after executing each dataset.
+	 *
+	 * @see org.openmrs.test.BaseContextSensitiveTest#executeDataSet(org.dbunit.dataset.IDataSet)
+	 */
+	@Override
+	public void executeDataSet(IDataSet dataset) throws Exception {
+		super.executeDataSet(dataset);
+		updateSearchIndex();
+	}
+	
+	@Test
+	@Verifies(value = "should save OrderSet", method = "save(OrderSet)")
+	public void shouldSaveOrderSet() throws Exception {
+		executeDataSet(ORDER_SET);
+		OrderSet orderSet = orderSetBuilder(false, false);
+		OrderSet orderSetObj = orderSetService.save(orderSet);
+		assertNotNull("OrderSet has a order_set_id", orderSetObj.getId().intValue());
```

Comment:
Do you need to call intValue() which returns a primitive when checking for not null?